src/renderer/entities/dynamicDerivations/model/import-keys-model.ts

```diff
@@ -0,0 +1,160 @@
+import { ImportErrorsLabel } from '@renderer/entities/dynamicDerivations/lib/constants';
+import { AccountId, ChainId, ObjectValues } from '@renderer/shared/core';
+
+type SampleFnError = { error: DerivationImportError };
+type ExistingDerivations = {
+  root: AccountId;
+  derivations: TypedImportedDerivation[];
+};
+type KeysImportReport = {
+  addedKeys: number;
+  updatedNetworks: number;
+  duplicatedKeys: number;
+  ignoredNetworks: ChainId[];
+};
+type MergeResult = { derivations: TypedImportedDerivation[]; report: KeysImportReport };
+
+const $validationError = createStore<ImportError | null>(null);
+const $successReport = createStore<KeysImportReport | null>(null);
+const $mergedKeys = createStore<TypedImportedDerivation[]>([]);
+
+const $existingDerivations = createStore<ExistingDerivations | null>(null);
+
+const fileUploaded = createEvent<string>();
+const importStarted = createEvent<ExistingDerivations>();
+
+const checkFileStructureFx = createEffect<string, ParsedImportFile, DerivationImportError>((fileContent: string) => {
+  // using default core scheme converts 0x strings into numeric values
+  const res = parse(fileContent, { schema: 'failsafe' });
+  if (importKeysUtils.isFileStructureValid(res)) {
+    return res;
+  } else {
+    throw new DerivationImportError(ImportErrorsLabel.INVALID_FILE_STRUCTURE);
+  }
+});
+
+const validateDerivationsFx = createEffect<ParsedImportFile, TypedImportedDerivation[], DerivationImportError>(
+  (result: ParsedImportFile) => {
+    const parsed = importKeysUtils.getDerivationsFromFile(result);
+    if (!parsed) {
+      throw new DerivationImportError(ImportErrorsLabel.INVALID_FILE_STRUCTURE);
+    }
+
+    const { derivations, root } = parsed;
+
+    if (root !== $existingDerivations.getState()?.root) {
+      throw new DerivationImportError(ImportErrorsLabel.INVALID_ROOT);
+    }
+
+    const invalidPaths: string[] = [];
+
+    derivations.forEach((derivation) => {
+      if (!derivation.derivationPath) {
+        throw new DerivationImportError(ImportErrorsLabel.INVALID_FILE_STRUCTURE);
+      }
+      if (!importKeysUtils.validateDerivation(derivation)) {
+        invalidPaths.push(derivation.derivationPath);
+      }
+    });
+
+    if (invalidPaths.length) {
+      if (invalidPaths.every((p) => p.includes('///'))) {
+        throw new DerivationImportError(ImportErrorsLabel.PASSWORD_PATH, invalidPaths);
+      } else {
+        throw new DerivationImportError(ImportErrorsLabel.INVALID_PATH, invalidPaths);
+      }
+    } else {
+      return derivations as TypedImportedDerivation[];
+    }
+  },
+);
+
+const mergePathsFx = createEffect((imported: TypedImportedDerivation[]): MergeResult | undefined => {
+  const mergeReport: KeysImportReport = {
+    addedKeys: 0,
+    updatedNetworks: 0,
+    duplicatedKeys: 0,
+    ignoredNetworks: [],
+  };
+  const mergeResult: TypedImportedDerivation[] = [];
+
+  const existingDerivations = $existingDerivations.getState()?.derivations;
```

Comment:
⚠️ `getState` should not be used [doc](https://effector.dev/docs/api/effector/store#getstate)

Reply:
changed effect to get existing derivations as input